File: Python/board/make_move.py
# -*- coding: utf-8 -*-
"""
Created on Sat Oct  1 19:32:43 2022

@author: Swen
"""
from common.names import DIRECTIONS

from .find_bracket import find_bracket
from .make_flip import make_flip
from .board_to_string import board_to_string

import time

def make_move(board, move, player, show_moves = False):
    """
        Place a brick on the 'move' square for player
    """
    board[move] = player

    # When the player makes a valid move, we need to update the board and flip all the
    # bracketed pieces. Look for a bracket in any direction
    for direction in DIRECTIONS:
        bracket = find_bracket(board, move, player, direction)
        if bracket:
            make_flip(board, move, player, direction, bracket)

    if show_moves:
        print(board_to_string(board))
        time.sleep(1)
    return board

File: Cpp/board/make_move.cc
#include "board.ih"

#include <iostream>

// Need to loop over the directions, so using a map instead of enum!
#include <unordered_map>
#include <string>
std::unordered_map<std::string, int> Directions 
{
    {"UP", -10},
    {"UP_RIGHT", -9},
    {"RIGHT",  1},
    {"DOWN_RIGHT", 11},
    {"DOWN",  10},
    {"DOWN_LEFT", 9},
    {"LEFT", -1},
    {"UP_LEFT", -11}
};

void make_move(int move, char player, char* board)
{
    board[move] = player;   // capture the move field

    for (std::pair<std::string, int> element : Directions)
    {
        int direction = element.second;
        //std::cout << direction << '\n';

        int bracket = find_bracket(move, player, board, direction);

        //std::cout << bracket << '\n';
        if (bracket != 0)
        {
            //std::cout << bracket << '\n';
            make_flip(move, player, board, direction, bracket);
        }
    }

    //print_board(board);
}